app/controllers/accounts_controller.rb

```diff
@@ -2,12 +2,10 @@ class AccountsController < ApplicationController
   before_action :authenticate_user!
 
   def new
-    if params[:type].blank? || Account.accountable_types.include?("Account::#{params[:type]}")
-      @account = if params[:type].blank?
-        Account.new
-      else
-        Account.new(accountable_type: "Account::#{params[:type]}")
-      end
+    type = params[:type].presence
+
+    if type.blank? || Account.accountable_types.include?("Account::#{type}")
+      @account = Account.new(accountable_type: "Account::#{type}")
     else
       head :not_found
     end
```

Comment:
Nice thinking! I think we can take it a step further.

Consider the following:

```rb
maybe(dev)> params = {}
maybe(dev)> params[:type] = "Foo"
maybe(dev)> "Account::#{params[:type]}".presence_in(Account.accountable_types).constantize
(irb):23:in `<main>': undefined method `constantize' for nil (NoMethodError)

"Account::#{params[:type]}".presence_in(Account.accountable_types).constantize
                                                                  ^^^^^^^^^^^^
maybe(dev)> params[:type] = "Credit"
maybe(dev)> "Account::#{params[:type]}".presence_in(Account.accountable_types).constantize
=> Account::Credit(id: uuid, created_at: datetime, updated_at: datetime)
```

We could use that to refactor as follows:
```rb
class AccountsController < ApplicationController
  before_action :authenticate_user!

  def new
    @account = Account.new accountable_type: account_type_class
  end

  private

  def account_type_class
    "Account::#{params[:type].presence_in(Account.accountable_types)}".constantize
  rescue NameError
    Account
  end
end
```

We're presumably okay with defaulting to `Account` per https://github.com/maybe-finance/maybe/blob/48ade396ca3a1a9142350ff4e4620bb80e1d2f75/app/controllers/accounts_controller.rb#L40

But if not, we can just raise the `NameError` — fine to throw an exception if we got this far without a valid type param. I'll let @Shpigford confirm.

Reply:
Actually, per https://github.com/maybe-finance/maybe/blob/48ade396ca3a1a9142350ff4e4620bb80e1d2f75/app/models/account.rb#L4

I don't think rescuing and returning `Account` makes sense. I think we should just raise.

Reply:
We just need to be careful as the new action renders a different partial depending on the presence of a valid type param.  I think I'll a quick test to make that doesn't break with refactors.  (But agree that this could be made cleaner)

Reply:
Well, it's a question of where the `.constantize` fails. If it won't fail, then `if..else` statement or a tenary operator is fine.

Reply:
Perhaps I need to look at this as a controller of its own rather than refactoring action by action. I also think a possible solution is to use `||` instead of raise even if `.constantize` happens to return a nit and fail.

Reply:
How is the type param obtained? Sorry I haven't looked myself. It shouldn't be open to freetext input. If we get to this point with an invalid type then someone's attempting something shady and it's fine to raise an exception. Arguably, even better to raise and capture in Sentry so we're aware of which attack vectors are being attempted.

Reply:
Via links rendered when the type parameter is blank